App/screens/CredentialOffer.tsx

```diff
@@ -31,123 +34,187 @@ const styles = StyleSheet.create({
   container: {
     flex: 1,
     flexDirection: 'column',
-    backgroundColor: CredentialOfferTheme.background,
+    backgroundColor: ColorPallet.brand.primaryBackground,
+  },
+  listItem: {
+    paddingHorizontal: 25,
+    paddingTop: 16,
+    backgroundColor: ColorPallet.brand.secondaryBackground,
+  },
+  label: {
+    ...TextTheme.label,
+  },
+  textContainer: {
+    justifyContent: 'space-between',
+    minHeight: TextTheme.normal.fontSize,
+    paddingVertical: 15,
+    height: 90,
+  },
+  text: {
+    ...TextTheme.normal,
   },
 })
 
 const CredentialOffer: React.FC<CredentialOfferProps> = ({ navigation, route }) => {
+  if (!route || !route.params) {
+    throw new Error('CredentialOffer route prams were not set properly')
+  }
+
+  const { credentialId } = route.params
   const { agent } = useAgent()
   const { t } = useTranslation()
+  const [, dispatch] = useContext(Context)
   const [buttonsVisible, setButtonsVisible] = useState(true)
   const [pendingModalVisible, setPendingModalVisible] = useState(false)
   const [successModalVisible, setSuccessModalVisible] = useState(false)
   const [declinedModalVisible, setDeclinedModalVisible] = useState(false)
+  const dateFormatOptions = { year: 'numeric', month: 'short', day: 'numeric' }
+  const credential = useCredentialById(credentialId)
 
-  if (!agent?.credentials) {
-    Toast.show({
-      type: ToastType.Error,
-      text1: t('Global.Failure'),
-      text2: t('Global.SomethingWentWrong'),
-    })
-
-    navigation.goBack()
-    return null
+  if (!credential) {
+    throw new Error('Unable to fetch credential from AFJ')
   }
 
-  const getCredentialRecord = (credentialId?: string): CredentialRecord | void => {
-    try {
-      if (!credentialId) {
-        throw new Error(t('CredentialOffer.CredentialNotFound'))
-      }
-      return useCredentialById(credentialId)
-    } catch (e: unknown) {
-      Toast.show({
-        type: ToastType.Error,
-        text1: t('Global.Failure'),
-        text2: t('Global.SomethingWentWrong'),
-      })
-
-      navigation.goBack()
-    }
+  if (!agent) {
+    throw new Error('Unable to fetch agent from AFJ')
   }
 
-  const credential = getCredentialRecord(route?.params?.credentialId)
+  const { name: schemaName } = parsedSchema(credential)
+  // @ts-ignore next-line
+  const { invitation } = useConnectionById(credential.connectionId)
 
-  if (!credential) {
-    Toast.show({
-      type: ToastType.Error,
-      text1: t('Global.Failure'),
-      text2: t('CredentialOffer.CredentialNotFound'),
-    })
-    navigation.goBack()
-    return null
+  if (!invitation) {
+    throw new Error('Unable to invitation from AFJ')
   }
 
   useEffect(() => {
+    if (credential.state === CredentialState.Declined) {
+      setDeclinedModalVisible(true)
+      return
+    }
+
     if (credential.state === CredentialState.CredentialReceived || credential.state === CredentialState.Done) {
```

Comment:
Isn't it better to keep the logic separated in different effects?

Reply:
Good question. I though because they both operate on the same state `}, [credential])` they were related enough to go into the same hook. If they had different state dependencies, or cleanup requirements, I'd probably separate them. If you think they're better separated I'm 100% ok with that.

Reply:
The way I look at it, because they are independent conditional branches we lose nothing by separating them.